src/layouts/index.js

```diff
@@ -0,0 +1,19 @@
+export function layout({layout: layout1, ...options}, layout2) {
+  if (layout2 == null) throw new Error("invalid layout");
+  layout2 = partialLayout(layout2);
+  if (layout1 != null) layout2 = composeLayout(layout1, layout2);
+  return {...options, layout: layout2};
+}
+
+function composeLayout(l1, l2) {
+  return function(index, scales, values, dimensions) {
+    values = l1.call(this, index, scales, values, dimensions);
```

Comment:
Under the redesign signature, a layout returns {index, values, channels}, so we don’t want to assign to *values* here. Similarly the implementation of partialLayout will need to change.